src/object.rs

```diff
@@ -309,6 +309,10 @@ glib_wrapper! {
 macro_rules! glib_object_wrapper {
     ([$($attr:meta)*] $name:ident, $ffi_name:path, $ffi_class_name:path, @get_type $get_type_expr:expr) => {
         $(#[$attr])*
+        // Always derive Debug/Hash (and below impl PartialEq, Eq) for object types. Due to
+        // inheritance and up/downcasting we must implement these by pointer or otherwise
+        // they would potentially give differeny results for the same object depending on
+        // the type we currently know for it
```

Comment:
Actually the same applies for `Ord` and `PartialOrd`. I think we should move those out of here too (I don't think there currently is any impl anyway).

In the future we might want to consider adding wrapper types for those that deref to the actual type and can only hold that type

Reply:
Agree that it good reason not implement most trait, but not sure about `Hash`

Reply:
Also IMHO we need add function for compare two `IsA<Object>` by compare pointers.

Reply:
> Agree that it good reason not implement most trait, but not sure about Hash

There's a section about `Hash` and its relation to `Eq` in the docs. Equal things need to have the same hash, which is obviously the case if the pointers are the same. But we would also have the same hash with different pointers and same "content" which is far from ideal, hash collisions :)



> Also IMHO we need add function for compare two IsA<Object> by compare pointers.

You mean `Ord`? Because `Eq` we have

Reply:
I meant `Eq` good it we have it already. I don't see it.

Reply:
https://github.com/gtk-rs/glib/blob/7d9217b313d0e2977df9f3528ec4e10877268bf1/src/object.rs#L566-L572 and https://github.com/gtk-rs/glib/blob/7d9217b313d0e2977df9f3528ec4e10877268bf1/src/object.rs#L598

I also added similar things for `Ord` in these commits now